src/Uno.UI/UI/Xaml/IFrameworkElement.cs

```diff
@@ -232,44 +238,36 @@ public static void InvalidateMeasure(this IFrameworkElement e)
 		}
 #endif
 
-		public static IFrameworkElement FindName(IFrameworkElement e, IEnumerable<View> subviews, string name)
+		public static IFrameworkElement FindName(IFrameworkElement e, ViewGroup group, string name)
 		{
 			if (string.Equals(e.Name, name, StringComparison.Ordinal))
 			{
 				return e;
 			}
 
-			var frameworkElements = subviews
-				.Safe()
-				.OfType<IFrameworkElement>()
-				.Reverse()
-				.ToArray();
```

Comment:
This was allocating lots of arrays and some internal LINQ iterators, as well as boxing MaterializableList enumerator.

![image](https://github.com/unoplatform/uno/assets/31348972/39f228f2-c74e-473b-a6a7-593dfd42a3dd)

Instead of getting an array of reversed subviews, we now just iterate over them in reversed order and search for what we want.